aws/src/main/java/org/apache/iceberg/aws/lakeformation/LakeFormationAwsClientFactory.java

```diff
@@ -132,27 +145,90 @@ private LakeFormationClient lakeFormation() {
+      this.cache = Caffeine.newBuilder()
+          .expireAfter(new Expiry<String, LakeFormationTemporaryCredentials>() {
+            @Override
+            public long expireAfterCreate(@NonNull String key,
+                @NonNull LakeFormationTemporaryCredentials value, long currentTime) {
+              return getExpiryDuration(value);
+            }
+
+            @Override
+            public long expireAfterUpdate(@NonNull String key, @NonNull LakeFormationTemporaryCredentials value,
+                long currentTime, @NonNegative long currentDuration) {
+              return getExpiryDuration(value);
+            }
+
+            @Override
+            public long expireAfterRead(@NonNull String key, @NonNull LakeFormationTemporaryCredentials value,
+                long currentTime, @NonNegative long currentDuration) {
+              return currentDuration;
+            }
+          })
+          .initialCapacity(1)
+          .build();
     }
 
     @Override
     public AwsCredentials resolveCredentials() {
-      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
-          GetTemporaryGlueTableCredentialsRequest.builder()
-              .tableArn(tableArn)
-              // Now only two permission types (COLUMN_PERMISSION and CELL_FILTER_PERMISSION) are supported
-              // and Iceberg only supports COLUMN_PERMISSION at this time
-              .supportedPermissionTypes(PermissionType.COLUMN_PERMISSION)
-              .build();
-      GetTemporaryGlueTableCredentialsResponse response =
-          client.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
-      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+      LakeFormationTemporaryCredentials credentials = cache.get(tableArn, key -> {
+        GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+            GetTemporaryGlueTableCredentialsRequest.builder()
+                .tableArn(key)
+                // Now only two permission types (COLUMN_PERMISSION and CELL_FILTER_PERMISSION) are supported
+                // and Iceberg only supports COLUMN_PERMISSION at this time
+                .supportedPermissionTypes(PermissionType.COLUMN_PERMISSION)
+                .build();
+        GetTemporaryGlueTableCredentialsResponse response =
+            client.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+
+        AwsSessionCredentials awsSessionCredentials = AwsSessionCredentials
+            .create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+
+        cachedCredentials = new LakeFormationTemporaryCredentials(awsSessionCredentials, response.expiration());
+        return cachedCredentials;
+      });
+
+      return credentials.credentials();
+    }
+
+    private long getExpiryDuration(LakeFormationTemporaryCredentials value) {
```

Comment:
In Iceberg, please avoid using getter https://github.com/apache/iceberg/blob/master/CONTRIBUTING.md#style